### PR TITLE
llvm12/DICompositeType: three new fields

### PR DESCRIPTION
**Pull request by dannypsnl**
* associated
* allocated
* rank

### CODE REVIEW

File: ll.tm [hunk 1]
```diff
@@ -4248,6 +4261,10 @@ ProducerField -> ProducerField
 	: 'producer:' Producer=StringLit
 ;
 
+RankField -> RankField
+	: 'rank:' Rank=MDSignedOrMDField
```
Comment (mewmew) [inline ll.tm]:
```suggestion
	: 'rank:' Rank=MDFieldOrInt
```

**Review by mewmew — COMMENTED** [review]
LGTM, with one minor edit. The `MDSignedOrMDField` field is called `MDFIeldOrInt` in the grammar. If we want, we can change the old occurrences to use `MDSignedOrMDField` and rename `MDFIeldOrInt` to `MDSignedOrMDField`.

For now, I've just used `MDFIeldOrInt` instead, to be consistent with `LowerBoundField`, `StrideField`, etc.

**Comment by dannypsnl** [comment]
@mewmew applied

**Review by mewmew — APPROVED** [review]
LGTM.

**Pull request merged**